crux-core/src/crux/index.clj

```diff
@@ -425,11 +352,23 @@
      relation)))
 
 (defn new-relation-virtual-index
-  ([relation-name tuples max-depth encode-value-fn]
-   (new-relation-virtual-index relation-name tuples max-depth encode-value-fn nil))
-  ([relation-name tuples max-depth encode-value-fn layered-range-constraints]
+  ([tuples max-depth encode-value-fn]
+   (new-relation-virtual-index tuples max-depth encode-value-fn nil))
+  ([tuples max-depth encode-value-fn layered-range-constraints]
    (update-relation-virtual-index! (->RelationVirtualIndex max-depth
                                                            (RelationVirtualIndexState. nil nil nil nil)
                                                            layered-range-constraints
                                                            encode-value-fn)
                                    tuples)))
+
+(defrecord SingletonVirtualIndex [v]
+  db/Index
+  (seek-values [_ k]
+    (when-not (pos? (mem/compare-buffers (or k mem/empty-buffer) v))
+      v))
+
+  (next-values [_]))
+
+(defn new-singleton-virtual-index [v encode-value-fn]
```

Comment:
nit: this fn feels like its implementation details are in the wrong place to me - save the detail of nippy's freeze fallback, we're only passing the encoded value through to the record. better done in query.clj as a helper function?